src/material/datepicker/date-range-input.ts

```diff
@@ -49,16 +59,14 @@ let nextUniqueId = 0;
   providers: [
     {provide: MatFormFieldControl, useExisting: MatDateRangeInput},
     {provide: MAT_DATE_RANGE_INPUT_PARENT, useExisting: MatDateRangeInput},
-
-    // TODO(crisbeto): this will be provided by the datepicker eventually.
-    // We provide it here for the moment so we have something to test against.
-    MAT_RANGE_DATE_SELECTION_MODEL_PROVIDER,
   ]
 })
 export class MatDateRangeInput<D> implements MatFormFieldControl<DateRange<D>>,
-  MatDateRangeInputParent, AfterContentInit, OnDestroy {
+  MatDatepickerControl<D>, MatDateRangeInputParent<D>, AfterContentInit, OnDestroy {
```

Comment:
should this component provide `NG_VALIDATORS` to check that start <= end? I think if used together with the recommended `FormGroup` approach that should work

Reply:
The way I was imagining it is that `MatStartDate` would have its own custom validator which checks that it doesn't come after the end date. The same goes for `MatEndDate`. That way you can have two separate error messages depending on which field is invalid.

Reply:
That sounds like it should work too. The only time it will come up is if someone is typing. If they select the dates backwards in the calendar we can probably just swap it for them